crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1342,6 +1363,40 @@ impl<'a, Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvide
     }
 }
 
+/// State that is used during unit propagation in the resolver.
+#[derive(Clone)]
+struct ResolverState {
+    /// The internal state used by the resolver.
+    ///
+    /// Note that not all parts of this state are strictly internal. For
+    /// example, the edges in the dependency graph generated as part of the
+    /// output of resolution are derived from the "incompatibilities" tracked
+    /// in this state. We also ultimately retrieve the final set of version
+    /// assignments (to packages) from this state's "partial solution."
+    pubgrub: State<UvDependencyProvider>,
+    /// The next package on which to run unit propgation.
+    next: PubGrubPackage,
+    /// The set of pinned versions we accrue throughout resolution.
+    ///
+    /// The key of this map is a package name, and each package name maps to
+    /// a set of versions for that package. Each version in turn is mapped
+    /// to a single `ResolvedDist`. That `ResolvedDist` represents, at time
+    /// of writing (2024/05/09), at most one wheel. The idea here is that
+    /// `FilePins` tracks precisely which wheel was selected during resolution.
+    /// After resolution is finished, this maps is consulted in order to select
+    /// the wheel chosen during resolution.
+    pins: FilePins,
+    /// When dependencies for a package are retrieved, this map of priorities
+    /// is updated based on how each dependency was specified. Certain types
+    /// of dependencies have more "priority" than others (like direct URL
+    /// dependencies). These priorities help determine which package to
+    /// consider next during resolution.
+    priorities: PubGrubPriorities,
+    /// This keeps track of the set of versions for each package that we've
+    /// already visited during resolution. This avoids doing redundant work.
+    added_dependencies: FxHashMap<PubGrubPackage, FxHashSet<Version>>,
```

Comment:
@charliermarsh What do you think about the comments here? Did I get things right? It would be useful to know if my understanding is correct. (Or even incomplete.)

Reply:
Yeah these look good.